langchain/chat/memory.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any, Dict, List, Optional
+
+from langchain.chains.base import Memory
```

Comment:
There's enough changes to all parts of memory in this pull, and `langchain.memory` is added, what do you think about moving the Memory classes out of `lanchain.chains.base` to `langchain.memory.base`  ?